frontend/src/components/common/Calendar/WeeklyCalendar.tsx

```diff
@@ -78,29 +78,71 @@ const DayButton = ({ day }: DayButtonProps) => {
   );
 };
 
-type RenderItemProps<EventItem> = {
-  item: Schedule | CheckIn;
-  showingFullWeek: boolean;
-  index: number;
+type RenderItemProps = {
+  item?: Schedule | CheckIn;
+  showingFullWeek?: boolean;
+  index?: number;
+  emptyState: boolean;
 };
 
-type WeeklyBodyProps<EventItem> = {
+type WeeklyBodyProps = {
   selectedDay: Date;
   items: Schedule[] | CheckIn[];
-  renderItem: (item: RenderItemProps<EventItem>) => ReactNode;
+  renderItem: (item: RenderItemProps) => ReactNode;
   handleDateChange: (days: number) => void;
   setSelectedDay: (date: Date) => void;
   calendarDate: Date;
 };
 
-export function WeeklyBody<EventItem>({
+const getFilteredDays = (
+  items: Array<Schedule | CheckIn>,
+  selectedDay: Date,
+  i: number,
+  renderItem: (item: RenderItemProps) => ReactNode,
+  week: Date,
+  locale?: Locale,
+) => {
+  const shiftsArr = items.filter((item) => {
+    const currentDate = setDay(week, selectedDay.getDay() + i, {
+      locale,
+    });
+    const scheduledDate =
+      "startTime" in item
+        ? new Date(item?.startTime as string)
+        : new Date(item?.startDate as string);
+    if (
+      item === null ||
+      scheduledDate === null ||
+      scheduledDate.getDate() !== currentDate.getDate() ||
+      scheduledDate.getMonth() !== currentDate.getMonth() ||
+      scheduledDate.getFullYear() !== currentDate.getFullYear()
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  return !shiftsArr.length
+    ? renderItem({ emptyState: true })
+    : shiftsArr.map((item, index) => {

```

Comment:
I'm seeing this crash consistently on this branch: 
![image](https://user-images.githubusercontent.com/19617248/164566877-fa7473de-470b-4df5-b096-2c4ab02ff5b4.png)

![image](https://user-images.githubusercontent.com/19617248/164566792-1efcf294-2e2e-451a-8967-3eb4ba7ccc05.png)